### PR TITLE
Bump ruff, drop `autoflake`, add `darglint` back

### PR DESCRIPTION
**Pull request by strickvl**
- Bumped ruff to the latest version (some bugfixes)
- dropped autoflake completely and replaced it with the corresponding `ruff` command
- added `darglint` back (and the associated `docstring.sh` script) because ruff wasn't fully covering our docstring checks (specifically, whether all args or exceptions raised were mentioned inside a docstring). thanks to @fa9r for catching that one!

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: .github/workflows/docstring.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Docstring Check
+
+on: workflow_call
+
+jobs:
+  spellcheck:
```
Comment (fa9r) [inline .github/workflows/docstring.yml:6]: misleading job name?

File: src/zenml/cli/utils.py [hunk 2]
```diff
@@ -1064,7 +1064,12 @@ def warn_unsupported_non_default_workspace() -> None:
 
 
 def print_page_info(page: Page[T]) -> None:
-    """Print all information pertaining to a page to show the amount of items and pages."""
+    """Print all information pertaining to a page to show the amount of items
+    and pages.
```
Comment (fa9r) [inline src/zenml/cli/utils.py:1068]: Two liners in the beginning were also disallowed by pydocstyle I remember, so I guess neither darglint nor ruff cover this one?

File: src/zenml/zen_stores/rest_zen_store.py [hunk 3]
```diff
@@ -600,8 +600,7 @@ def list_flavors(
 
         Args:
             flavor_filter_model: All filter parameters including pagination
-            params
-
+                params
```
Comment (fa9r) [inline src/zenml/zen_stores/rest_zen_store.py:603]: ah this finally detects wrongly formatted docstrings again, nice 🙏

**Review by fa9r — APPROVED** [review]
Looks good, and thanks a lot for fixing all these docstrings again!

File: .github/workflows/docstring.yml (hunk 1 above)
Comment (AlexejPenner) [inline .github/workflows/docstring.yml:6]:
```suggestion
  docstring:
```

File: src/zenml/cli/utils.py (hunk 2 above)
Comment (strickvl) [reply src/zenml/cli/utils.py:1068]: This was caught by the linting check. Fixed now, but Ruff is catching this.

**Pull request merged**